### ISSUE TITLE
Add high level llama-cpp-python logits processor

### ISSUE THREAD

**Issue by dtiarks** [outlines-dev/outlines#540]
Implement an outlines based logits processor for llama-cpp-python according to the interface in docs: https://llama-cpp-python.readthedocs.io/en/latest/api-reference/#llama_cpp.LogitsProcessor

**Comment by rlouf** [comment]
This would be a much better way to integrate with llama.cpp since it would require less maintenance on our end.

**Comment by dtiarks** [comment]
Yeah, I guess this is also true for transformers and others (👀 TensorRT-LLM).
